dpctl/tensor/_manipulation_functions.py

```diff
@@ -727,12 +727,16 @@ def result_type(*arrays_and_dtypes):
              The dtype resulting from an operation involving the
              input arrays and dtypes.
     """
-    dtypes = [
-        X.dtype if isinstance(X, dpt.usm_ndarray) else dpt.dtype(X)
-        for X in arrays_and_dtypes
-    ]
-
-    _supported_dtype(dtypes)
+    dtypes = []
+    for X in arrays_and_dtypes:
+        if isinstance(X, dpt.usm_ndarray):
+            dtypes.append(X.dtype)
+        elif np.isscalar(X):
+            dtypes.append(X)
```

Comment:
This branch is not necessary, as scalars are treated in the fall-through clause:

```
In [1]: import numpy as np

In [2]: np.float32(2)
Out[2]: 2.0

In [3]: np.dtype(_)
Out[3]: dtype('float32')
```

Reply:
I see that I not correct for Python scalars though.